server/server.go

```diff
@@ -243,10 +260,23 @@ func NewServer(userConfig UserConfig, config Config) (*Server, error) {
 	}
 	lockingClient := locking.NewClient(boltdb)
 	workingDirLocker := events.NewDefaultWorkingDirLocker()
-	workingDir := &events.FileWorkspace{
+
+	var workingDir events.WorkingDir = &events.FileWorkspace{
 		DataDir:       userConfig.DataDir,
 		CheckoutMerge: userConfig.CheckoutStrategy == "merge",
 	}
+	// provide fresh tokens before clone from the GitHub Apps integration, proxy workingDir
+	if githubAppEnabled {
+		if !userConfig.WriteGitCreds {
```

Comment:
I've changed this in my branch and instead will require users to set --write-git-creds. I didn't want to assume users wanted that file written, especially because it clobbers the old file. I've also changed that a bit in my branch so it just replaces the line that needs replacing.